### ISSUE TITLE
The `ScrollNotification` parameter type used in the notification callback of `ScrollNotificationObserver` is confusing

### ISSUE THREAD

**Issue by 0xZOne** [flutter/flutter#126404]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

sdfasd

### Expected results

dasf

### Actual results

asdf

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by maRci002** [comment]
This is because `_ConvertedScrollMetricsNotification` extends `ScrollUpdateNotification`

```dart
class _ConvertedScrollMetricsNotification extends ScrollUpdateNotification { ... }
```

Change `if (notification is ScrollUpdateNotification) {` to `if (notification.runtimeType == ScrollUpdateNotification) {`

**Comment by 0xZOne** [comment]
> This is because `_ConvertedScrollMetricsNotification` extends `ScrollUpdateNotification`
> 
> ```dart
> class _ConvertedScrollMetricsNotification extends ScrollUpdateNotification { ... }
> ```
> 
> Change `if (notification is ScrollUpdateNotification) {` to `if (notification.runtimeType == ScrollUpdateNotification) {`

Yeah, It is confusing that `_ConvertedScrollMetricsNotification` is a `ScrollUpdateNotification`.

**Comment by TahaTesser** [comment]
I noticed this behavior when investigating fix for  https://github.com/flutter/flutter/issues/120083

cc: @Piinks
